index.bs

```diff
@@ -1370,77 +880,818 @@ What configuration is the device in after it resets?
+            1. <a>For each</a> |configuration| of {{USBDevice/[[configurations]]}}:
+                1. If |configuration|.{{USBConfiguration/configurationValue}} is equal to 
+                    |configurationValue|, set |selectedConfiguration| to |configuration|
+                    and break.
+            1. If |selectedConfiguration| is <code>null</code>, <a>reject</a> |promise| 
+                with a {{NotFoundError}} and abort these steps.
+            1. If <code>|device|.{{USBDevice/opened}}</code> is not equal to
+                <code>true</code> <a>reject</a> |promise| with an {{InvalidStateError}} and
+                abort these steps.
+            1. If |device|.{{USBDevice/configuration}} is not <code>null</code>:
+                1. <a>For each</a> |interface| of
+                    |device|.{{USBDevice/configuration}}.{{USBConfiguration/interfaces}}:
+                    1. Perform <a>to abort transfers currently scheduled on an interface</a> with |interface|.
+            1. Issue a <code>SET_CONFIGURATION</code> <a>control transfer</a> with
+                <code>configurationValue</code> set to the |configurationValue|.
+                If this step fails <a>reject</a> |promise| with a {{NetworkError}} and
+                abort these steps.
+            1. Let |numInterfaces| be the <a>size</a> of |selectedConfiguration|.{{USBConfiguration/interfaces}}.
+            1. Resize {{USBDevice/[[selectedAlternateSetting]]}} to |numInterfaces|.
+            1. Fill {{USBDevice/[[selectedAlternateSetting]]}} with 0.
+            1. Resize {{USBDevice/[[claimedInterface]]}} to |numInterfaces|.
+            1. Fill {{USBDevice/[[claimedInterface]]}} with <code>false</code>.
+            1. Set {{USBDevice/[[configurationValue]]}} to |configurationValue| 
+                and <a>resolve</a> |promise|.
+        </div>
+
+    : <dfn>claimInterface(interfaceNumber)</dfn>
+    ::
+        <div algorithm="claimInterface">
+            When invoked, MUST return a new {{Promise}} |promise| and run the following 
+            steps <a>in parallel</a>:
+                1. Let |interfaceNumber| be the parameter of <code>interfaceNumber</code>.
+                1. Let |device| be the target {{USBDevice}} object.
+                1. If |device| is no longer connected to the system, <a>reject</a> |promise|
+                    with a {{NotFoundError}} and abort these steps.
+                1. if |device|.{{USBDevice/configuration}} is <code>null</code>, <a>reject</a> |promise|
+                    with a {{InvalidStateError}} and abort these steps.
+                1. Let |interfaces| be |device|.{{USBDevice/configuration}}.{{USBConfiguration/interfaces}}.
+                1. Let |interfaceIndex| be the result of <a>to find the interface index</a> with
+                    |interfaceNumber| and |device|.{{USBDevice/configuration}}.
+                1. If |interfaceIndex| is equal to |interfaces|'s <a>size</a>,
```

Comment:
Can you have "find the interface index" return a sentinel value like -1 to indicate that the interface was not found? As written the algorithm doesn't make it totally clear what value is returned in that case and comparing it to the size of the list takes a bit to understand the intent.

Reply:
Updated in https://github.com/WICG/webusb/pull/212/commits/1a96f493fc285da99d2b812d809c241388197beb.